Bottom: d24c22bc8c6bb8268717e15c654972f5fe05c929
Top:    afa136149cd8a9a77132afbd988705da3c7fc59d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 19:54:09 +0000

polygon1.c: Refactor poly_Copy0 to use poly_Create(), rather than open-coding

This ensures all polygon initialisation happens in a common place


---

diff --git a/src/polygon1.c b/src/polygon1.c
index a8d9391..054bef4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2816,12 +2816,10 @@ poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
   if (src == NULL)
     return TRUE;
 
-  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
